hydra_base/lib/users.py

```diff
@@ -174,6 +176,54 @@ def delete_user(deleted_user_id,**kwargs):
 
     return 'OK'
 
+def get_failed_login_count(username, **kwargs):
+    try:
+        user_i = db.DBSession.query(User).filter(User.username == username).one()
+    except NoResultFound:
+        """ Non-existent user should not raise """
+        return -1
+
+    return user_i.failed_logins
+
+
+def get_max_login_attempts(*args, **kwargs):
+    max_login_attempts = int(config.get("security", "max_login_attempts"))
+
+    return max_login_attempts if max_login_attempts is not None else 0
```

Comment:
With 0 max login attempts, wouldn't this result in it failing always because the user would never be allowed to log in?

Reply:
Following discussion, 488f41f updates to fail on import if `max_login_attempts` not defined in config.
This getter is changed to use and explicit default and its behaviour is documented.